docs/YARN.md

```diff
@@ -0,0 +1,294 @@
+- `yarn workspaces foreach --exclude my-pattern/* run <executable>` in worktree
+  -> OK if dependency is not specified in workspace as long as it is in the
+  worktree
+
+See https://yarnpkg.com/features/workspaces#what-does-it-mean-to-be-a-workspace
+
+## Specify local dependencies as workspaces
+
+I.e. `workspace:packages/xxx`
+
+https://yarnpkg.com/features/workspaces#workspace-ranges-workspace
+
+## Adjust lifecycle scripts
+
+Unlike npm or Yarn v1, Yarn v2 won’t run arbitrary lifecycle scripts (i.e.
+`prexxx` and `postxxx`). There are
+[some exceptions](https://yarnpkg.com/advanced/lifecycle-scripts), including
+`postinstall`, but if you’ve been relying on any other lifecycle scripts you
+will need to
+[explicitly call them](https://yarnpkg.com/getting-started/migration#explicitly-call-the-pre-and-post-scripts)
+in the main script definition.
+
+> NOTE: Yarn
+> [recommends explicitly calling the lifecycle scripts](https://yarnpkg.com/getting-started/migration#explicitly-call-the-pre-and-post-scripts)
+> in the main script, but I would recommend not doing that, because it’s going
+> to get confusing if people start trying to use other package managers. It’s
+> much simpler just to put it all in the same script definition, or outsource to
+> a script file.
+
+## Check it works
+
+At this point you should have a working setup.
+
+## Remove shx or equivalents (probably)
+
+According to
+[this blog post](https://dev.to/arcanis/introducing-yarn-2-4eh1#normalized-shell):
+
+> Yarn 2 ships with a rudimentary shell interpreter that knows just enough to
+> give you 90% of the language structures typically used in the scripts field.
+> Thanks to this interpreter, your scripts will run just the same regardless of
+> whether they're executed on OSX or Windows.
+
+So you probably don’t need `shx` or any other tool that you were using for
+cross-platform shell compatibility. Then again, maybe you do still need it
+because you use a language structure outside the 90%. 🤷
+
+## Consider using constraints
+
+E.g. to ensure that each package specifies the right fields in its
+`package.json`. Bonus: you get to learn Prolog!
+
+https://yarnpkg.com/features/constraints
+
+## TypeScript plugin
+
+Automatically installs `@types` definitions if the project doesn’t have its own.
+
+```sh
+yarn plugin import typescript
+```
+
+https://github.com/yarnpkg/berry/tree/master/packages/plugin-typescript
+
+## Version plugin
+
+```sh
+yarn plugin import version
+```
+
+## interactive-tools plugin
+
+Most useful for `yarn upgrade-interactive`
+https://yarnpkg.com/cli/upgrade-interactive
+
+## Plug-n-play and node_modules
+
+Read about the problems of `node_modules` and how Yarn aims to fix them here:
+https://yarnpkg.com/features/pnp#the-node_modules-problem
+
+### Check whether your project is ready
+
+```sh
+yarn dlx @yarnpkg/doctor@2
+```
+
+> NOTE: You need to specify the `@2` because they released the incompatible v3
+> RC as `latest` on npm.
```

Comment:
Why format this as quote?

Reply:
I also don't like it, but there isn't a good information box in GFMD as far as I can tell. People seem to use either this or a horrible little table.

Reply:
What about a simple paragraph starting with bold, like this:

**NOTE:** You need to specify the `@2` because they released the incompatible v3 RC as `latest` on npm.